bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeChecker.java

```diff
@@ -1509,7 +1509,7 @@ private static boolean checkFillerValue(BObjectType type) {
             AttachedFunction initializerFunc = type.initializer;
             if (initializerFunc == null) {
                 // No __init function found.
-                return true;
+                return !Flags.isFlagOn(Flags.ABSTRACT, type.flags);
```

Comment:
Doesn't the initializerFunc become null for abstract object types only?

Reply:
Yep, removed redundant check.